src/WebJobs.Script/ExtensionBundle/ExtensionBundleManager.cs

```diff
@@ -293,6 +294,38 @@ internal static string FindBestVersionMatch(VersionRange versionRange, IEnumerab
             return matchingVersion?.ToString();
         }
 
+        private NuGetVersion ResolvePlatformReleaseChannelVersion(IEnumerable<NuGetVersion> bundleVersions, NuGetVersion currentMatchingVersion)
+        {
+            bool isStandardOrExtendedChannel = string.Equals(_platformReleaseChannel, ScriptConstants.StandardPlatformChannelName, StringComparison.Ordinal) ||
+                                   string.Equals(_platformReleaseChannel, ScriptConstants.ExtendedPlatformChannelName, StringComparison.Ordinal);
+
+            bool isLatestOrEmptyChannel = string.Equals(_platformReleaseChannel, string.Empty, StringComparison.Ordinal) ||
+                                          string.Equals(_platformReleaseChannel, ScriptConstants.LatestPlatformChannelName, StringComparison.Ordinal);
+
+            if (isStandardOrExtendedChannel)
+            {
+                if (bundleVersions.Count() > 1)
+                {
+                   return bundleVersions.OrderByDescending(version => version.Version).ElementAt(1); // if n is latest, should get version n-1
+                }
+                else
+                {
+                    // keep the latest version, log a notice
+                    _logger.LogInformation("Unable to apply plaform release channel configuration {platformReleaseChannelName}. Only one matching bundle version is available. {latestBundleVersion} will be used", _platformReleaseChannel, currentMatchingVersion);
+                }
+            }
+            else if (isLatestOrEmptyChannel)
+            {
+                // no - op, latest version should be used.
```

Comment:
Shouldn't you be returning `return bundleVersions.OrderByDescending(version => version.Version).FirstOrDefault();` here?

Reply:
Maybe we can move line 266 to this method. So, there is a single method which determines what version to use.